sympy/simplify/simplify.py

```diff
@@ -1569,32 +1571,64 @@ def _reduce_inverses(_args):
         inverses = []
         args = []
         for arg in _args:
-            if isinstance(arg, Pow) and arg.args[1] < 0:
+            if isinstance(arg, _Pow) and arg.args[1] < 0:
```

Comment:
What is `_Pow`? I can't find where this is defined.

Reply:
Ah, it's actually defined later on lines 1610-1612. I hadn't noticed until now but Python doesn't seem to mind. `_Pow` is `Pow` for symbolic expressions and `MatPow` for matrix expressions, similarly for `_Mul`, `_Add` and `_Symbol`

Reply:
I see. That would be much clearer if that line were moved to the top of the `nc_simplify` function, before the inner functions are defined.